framework/global/ptrutils.h

```diff
@@ -0,0 +1,55 @@
+//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//
+//  Copyright (C) 2020 MuseScore BVBA and others
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program; if not, write to the Free Software
+//  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+//=============================================================================
+#ifndef MU_FRAMEWORK_PTRUTILS_H
+#define MU_FRAMEWORK_PTRUTILS_H
+
+#include "runtime.h"
+#include "log.h"
+
+namespace mu {
+namespace ptr {
+template<typename T, typename E> T* checked_cast(E* source)
+{
+#ifndef NDEBUG
+    T* casted = dynamic_cast<T*>(source);
```

Comment:
>> Is there any reason to not use only static_cast?

Because

```
    struct A
    {
        virtual ~A() = default;
    };

    struct B : public A
    {
    };

    struct C : public A
    {
    };

    A* c = new C();

    B* b_c = static_cast<B*>(c);
    B* b_c_dyn = dynamic_cast<B*>(c);

    LOGI() << "b_c: " << b_c << ", b_c_dyn: " << b_c_dyn;
```

output: `b_c:  0x561600215de0 , b_c_dyn:  0x0`

>> In general, I thought RTTI is disabled within the editor
As far as I can see - no, at least in the debug RTTI is turned on.
Although I don't understand why turn it off, an attempt to increase performance ... hmm.